test/hotspot/jtreg/compiler/c2/irTests/TestMinMaxIdeal.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2025 IBM Corporation. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8373396
+ * @summary Verify that min/max add ideal optimizations get applied correctly
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib /
+ * @run driver compiler.c2.irTests.TestMinMaxIdeal
+ */
+
+package compiler.c2.irTests;
+
+import compiler.lib.compile_framework.CompileFramework;
+import compiler.lib.template_framework.Template;
+import compiler.lib.template_framework.TemplateToken;
+import compiler.lib.template_framework.library.CodeGenerationDataNameType;
+import compiler.lib.template_framework.library.PrimitiveType;
+import compiler.lib.template_framework.library.TestFrameworkClass;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static compiler.lib.template_framework.Template.let;
+import static compiler.lib.template_framework.Template.scope;
+
+public class TestMinMaxIdeal {
+    public static void main(String[] args) {
+        // Create a new CompileFramework instance.
+        CompileFramework comp = new CompileFramework();
+
+        String templatedPackage ="compiler.c2.templated";
+        String templatedClassName ="MinMaxIdeal";
+        String templatedFQN = "%s.%s".formatted(templatedPackage, templatedClassName);
```

Comment:
That looks a bit convoluted. Why not just use the final string?

Reply:
Not sure exactly what you mean. For `addJavaSourceCode` I need the combined FQN and for `TestFrameworkClass.render` I need the classname and package separated.

Reply:
It is just a nit, I leave it to you. I'm also fine taking it as is :)